Source/Fuse.Elements/Element.Batching.uno

```diff
@@ -47,24 +45,32 @@ namespace Fuse.Elements
 
 		protected void DrawUnderlayChildren(DrawContext dc)
 		{
-			for (int i = 0; i < _firstNonUnderlay; i++)
-				ZOrder[i].Draw(dc);
+			var zOrder = GetCachedZOrder();
+			for (var i = 0; i < zOrder.Length; i++) 
+			{
+				var v = zOrder[i];
+				if (v.Layer != Layer.Underlay) return;
+				v.Draw(dc);
+			}
 		}
 
 		protected void DrawNonUnderlayChildren(DrawContext dc)
 		{
 			if (!HasChildren) return;
 
-			EnsureSortedZOrder();
-
+			var zOrder = GetCachedZOrder();
 			if (!ShouldBatch())
 			{
 				// get rid of old element batcher
 				if (_elementBatcher != null)
 					_elementBatcher = null;
 
-				for (int i = _firstNonUnderlay; i < ZOrder.Count; ++i)
-					ZOrder[i].Draw(dc);
+				for (var i = 0; i < zOrder.Length; i++) 
+				{
+					var v = zOrder[i];
+					if (v.Layer == Layer.Underlay) continue;
```

Comment:
I don't understand what you mean here. The `DrawNonUnderlayChildren` method is not supposed to draw underlays. That's what the code means.